ci-operator/step-registry/azure/provision/resourcegroup/cross-subscription/azure-provision-resourcegroup-cross-subscription-commands.sh

```diff
@@ -0,0 +1,82 @@
+REGION="${HYPERSHIFT_AZURE_LOCATION:-${LEASED_RESOURCE}}"
+echo "Azure region: ${REGION}"
+
+# az should already be there
+command -v az
+az --version
+
+# set the parameters we'll need as env vars
+if [[ -f "${SHARED_DIR}/azure-sp-contributor.json" ]]; then
+  echo "Setting AZURE credential with Contributor role only for installer"
+  AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/azure-sp-contributor.json" 
+else
+  echo "Setting AZURE credential with  default SP for installer"
+  AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+fi
+if [[ "${USE_HYPERSHIFT_AZURE_CREDS}" == "true" ]]; then
+  AZURE_AUTH_LOCATION="/etc/hypershift-ci-jobs-azurecreds/credentials.json"
+fi
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+#AZURE_AUTH_SUBSCRIPTION_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .subscriptionId)"
+# the crossSubscription ("Openshift QE 1") is saved in azure-sp-contributor.json 
+CROSS_SUBSCRIPTION_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .crossSubscriptionId)"
+echo "${CROSS_SUBSCRIPTION_ID}" > "${SHARED_DIR}/cross_subscription_id" 
+
+# log in with az
+if [[ "${CLUSTER_TYPE}" == "azuremag" ]]; then
+    az cloud set --name AzureUSGovernment
+elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
+    if [ ! -f "${CLUSTER_PROFILE_DIR}/cloud_name" ]; then
+        echo "Unable to get specific ASH cloud name!"
+        exit 1
+    fi
+    cloud_name=$(< "${CLUSTER_PROFILE_DIR}/cloud_name")
+
+    AZURESTACK_ENDPOINT=$(cat "${SHARED_DIR}"/AZURESTACK_ENDPOINT)
+    SUFFIX_ENDPOINT=$(cat "${SHARED_DIR}"/SUFFIX_ENDPOINT)
+
+    if [[ -f "${CLUSTER_PROFILE_DIR}/ca.pem" ]]; then
+        cp "${CLUSTER_PROFILE_DIR}/ca.pem" /tmp/ca.pem
+        cat /usr/lib64/az/lib/python*/site-packages/certifi/cacert.pem >> /tmp/ca.pem
+        export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+    fi
+    az cloud register \
+        -n ${cloud_name} \
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+az account set --subscription ${CROSS_SUBSCRIPTION_ID}
+
+# create an empty resource group
+az group create -l "${REGION}" -n "${RG_NAME}"
+
+# save resource group information to ${SHARED_DIR} for reference and deprovision step
+echo "${RG_NAME}" > "${SHARED_DIR}/resourcegroup_cross-sub"
+
+# Grant the Contributor role to service principal on the resource group in cross subscription 
+#az role assignment create \
+#  --role "Contributor" \
+#  --assignee "${AZURE_AUTH_CLIENT_ID}" \
+#  --scope /subscriptions/"${CROSS_SUBSCRIPTION_ID}"/resourceGroups/"${RG_NAME}"  
```

Comment:
Is it because that azure-sp-contributor sp is assigned "Contributor" role on scope of cross subscription level so we don't need to assign role again here?

Reply:
Yes, you are right, I just try to keep the code in case we could implement it with different way :)